File: src/main/kotlin/com/wynnlab/spells/archer/ArcherMain.kt
package com.wynnlab.spells.archer

import com.wynnlab.random
import com.wynnlab.spells.Spell
import com.wynnlab.spells.SpellData
import org.bukkit.Material
import org.bukkit.Sound
import org.bukkit.entity.Arrow
import org.bukkit.entity.Player
import org.bukkit.entity.Snowball
import org.bukkit.inventory.ItemStack

class ArcherMain(player: Player) : Spell(player, 1, SpellData.METEOR) {
    override fun tick() {
        player.playSound(player.location, Sound.ENTITY_ARROW_SHOOT, 1f, 1f)
        if (clone) {
            val arrow = player.launchProjectile(Snowball::class.java, player.eyeLocation.direction.clone().multiply(3))
            arrow.addScoreboardTag("arrow")
            arrow.item = ItemStack(Material.FLINT)
            arrow.setRotation(random.nextFloat(), 0f)
        } else {
            val arrow = player.launchProjectile(Arrow::class.java, player.eyeLocation.direction.clone().multiply(3))
            arrow.addScoreboardTag("arrow")
        }
    }
}

File: src/main/resources/classes/archer/archer_main.py
from org.bukkit import Material, Sound
from org.bukkit.entity import Arrow, Player, Snowball
from org.bukkit.inventory import ItemStack

from com.wynnlab.spells import PySpell

class Spell(PySpell):
    def tick(self):
        self.sound(Sound.ENTITY_ARROW_SHOOT, 1, 1)

        arrow = self.player.launchProjectile(Snowball if self.clone else Arrow, self.player.getEyeLocation().getDirection().multiply(3))
        if self.clone:
            arrow.setItem(ItemStack(Material.FLINT))

        arrow.addScoreboardTag('arrow')
        arrow.setShooter(self.player)

def delete_arrow(event):
    hit = event.getHitEntity()
    if not hit is None and not isinstance(hit, Player):
        PySpell.damage(event.getEntity().getShooter(), hit, True, 1)
        PySpell.knockbackFromPlayer(hit, event.getEntity().getShooter(), .5)

    event.getEntity().remove()

PySpell.registerProjectileHit('arrow', delete_arrow)